### ISSUE TITLE
Variational Inference Techniques in Deep Learning — 平芜泫的幻想乡

### ISSUE THREAD

**Issue by korepwx** [korepwx/blog-comments#1]
https://ipwx.me/2017/11/24/variational-inference/ 

 OverviewSuppose there are two random variables ${ {\boldsymbol{\mathrm{ {x} } } } }$ and ${ {\boldsymbol{\mathrm{ {z} } } } }$, where ${ {\boldsymbol{\mathrm{ {x} } } } }$ is observed and ${ {\boldsymbol{\mathrm{ {z} } } } }$ is not. We assume these two variables are modeled with a neural network, with parameter ${ {\boldsymbol{\mathrm{ {\theta} } } } }$. The prior of ${ {\boldsymbol{\mathrm{ {z} } } } }$ is $p _ {\theta}({ {\boldsymbol{\mathrm{ {z} } } } })$, while the posterior of ${ {\boldsymbol{\mathrm{ {x} } } } }$ given ${ {\boldsymbol{\mathrm{ {z} } } } }$ is $p _ {\theta}({ {\boldsymbol{\mathrm{ {x} } } } } \vert { {\boldsymbol{\mathrm{ {z} } } } })$. Although computing $p _ {\theta}({ {\boldsymbol{\mathrm{ {x} } } } } \vert { {\boldsymbol{\mathrm{ {z} } } } })$ is straightforward in this scenario, the posterior of ${ {\boldsymbol{\mathrm{ {z} } } } }$ given ${ {\boldsymbol{\mathrm{ {x} } } } }$, i.e., $p _ {\theta}({ {\boldsymbol{\mathrm{ {z} } } } } \vert { {\boldsymbol{\mathrm{ {x} } } } })$, is intractable by analytic methods.While it is possible to use sampling based methods to evaluate $p _ {\theta}({ {\boldsymbol{\mathrm{ {x} } } } })$ and further obtain $p _ {\theta}({ {\boldsymbol{\mathrm{ {z} } } } } \vert { {\boldsymbol{\mathrm{ {x} } } } }) = p _ {\theta}({ {\boldsymbol{\mathrm{ {x} } } } },{ {\boldsymbol{\mathrm{ {z} } } } }) / p _ {\theta}({ {\boldsymbol{\mathrm{ {x} } } } })$, it often demands a large number of samples. Alternatively, one may fit another neural network $q _ {\phi}({ {\boldsymbol{\mathrm{ {z} } } } } \vert { {\boldsymbol{\mathrm{ {x} } } } })$ to approximate the true posterior $p _ {\theta}({ {\boldsymbol{\mathrm{ {z} } } } } \vert { {\boldsymbol{\mathrm{ {x} } } } })$, by using the variational inference techniques.In the context of deep learning, a variational inference algorithm that fits $q _ {\phi}({ {\boldsymbol{\mathrm{ {z} } } } } \vert { {\boldsymbol{\mathrm{ {x} } } } })$ should basically have two elements. One is the training objective, which we may call variational objective hereafter. The other is the gradient estimator, since almost the entire deep learning is based on gradient descent techniques.
